cmd/trace-agent/subcommands/run/command.go

```diff
@@ -20,6 +21,10 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+type contextSupplier struct {
+	ctx context.Context
+}
+
```

Comment:
Yes, it's an fx thing. fx uses the most specific type available, which in this case is `context.cancelCtx` which isn't exported. I think I got it working with `fx.Provide` but since fx makes everything a singleton I didn't want to lock into one context.